src/wp-includes/block-template-utils.php

```diff
@@ -1477,3 +1477,30 @@ function inject_ignored_hooked_blocks_metadata_attributes( $post, $request ) {
 	$post->post_content = $content;
 	return $post;
 }
+
+/**
+ * Inject ignoredHookedBlocks metadata attributes into a template or template part before we return
+ * it in the REST API response.
+ *
+ * @since 6.6.0
+ * @access private
+ *
+ * @param WP_REST_Response  $response  The response object.
+ * @param WP_Block_Template $template  An object representing a template or template part
+ *
+ * @return WP_Block_Template The updated object representing a template or template part.
+ */
+function inject_ignored_hooked_blocks_metadata_attributes_for_response( $response, $template ) {
+	$hooked_blocks = get_hooked_blocks();
+	if ( empty( $hooked_blocks ) && ! has_filter( 'hooked_block_types' ) ) {
+		return $template;
+	}
+
+	$before_block_visitor = make_before_block_visitor( $hooked_blocks, $template, 'set_ignored_hooked_blocks_metadata' );
+	$after_block_visitor  = make_after_block_visitor( $hooked_blocks, $template, 'set_ignored_hooked_blocks_metadata' );
+
+	$blocks = parse_blocks( $template->content );
+	$template->content = traverse_and_serialize_blocks( $blocks, $before_block_visitor, $after_block_visitor );
+
+	return $template;
```

Comment:
I'm a bit surprised that we're returning `$template` 🤔 Aren't all WP filters expected to return the (potentially modified) _first_ argument -- i.e. the `$response`, in this case?

Reply:
Yeah, you're absolutely right. This PR hadn't gone through any testing (or had any test coverage) yet which is where I'd have likely caught this but thanks for flagging it here!

Reply:
https://github.com/WordPress/wordpress-develop/pull/6330/commits/04435c525a27e552a926cc4de241e743cb3497c1